Move UnsafeTensorFromTH and UnsafeStorageFromTH off Type

li-roy opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #21964 Remove Type dispatch
* #21963 Stop using Type in Python bindings
* #21941 Remove many usages of Type
* #21940 Remove getDeviceFromPtr and allocator from Type
* **#21923 Move UnsafeTensorFromTH and UnsafeStorageFromTH off Type**
* #21320 [BC-BREAKING] Replace Type dispatch with ATenDispatch

In preparation to delete Type.

Differential Revision: [D15883390](https://our.internmc.facebook.com/intern/diff/D15883390)

Commit:
Move UnsafeTensorFromTH and UnsafeStorageFromTH off Type

Commit:
Update on "Move UnsafeTensorFromTH and UnsafeStorageFromTH off Type"

Move UnsafeTensorFromTH and UnsafeStorageFromTH off Type

gh-metadata: pytorch pytorch 21923 gh/li-roy/28/head

Commit:
Update on "Move UnsafeTensorFromTH and UnsafeStorageFromTH off Type"

Move UnsafeTensorFromTH and UnsafeStorageFromTH off Type

gh-metadata: pytorch pytorch 21923 gh/li-roy/28/head

Commit:
Update on "Move UnsafeTensorFromTH and UnsafeStorageFromTH off Type"

Move UnsafeTensorFromTH and UnsafeStorageFromTH off Type

gh-metadata: pytorch pytorch 21923 gh/li-roy/28/head

gchanan commented on aten/src/ATen/core/UnsafeFromTH.h:
```diff
@@ -0,0 +1,20 @@
+
+namespace at {
+
+inline Tensor unsafeTensorFromTH(void * th_pointer, bool retain) {
```
maybe we can just put this in at::internal or at::detail or something?  I don't think anyone besides us should be calling this.

gchanan reviewed (approved):
I'd prefer making the namespace scarier, but otherwise looks good.

Commit:
Update on "Move UnsafeTensorFromTH and UnsafeStorageFromTH off Type"

Move UnsafeTensorFromTH and UnsafeStorageFromTH off Type

gh-metadata: pytorch pytorch 21923 gh/li-roy/28/head

Commit:
Update on "Move UnsafeTensorFromTH and UnsafeStorageFromTH off Type"

Move UnsafeTensorFromTH and UnsafeStorageFromTH off Type

gh-metadata: pytorch pytorch 21923 gh/li-roy/28/head

Commit:
Update on "Move UnsafeTensorFromTH and UnsafeStorageFromTH off Type"

Move UnsafeTensorFromTH and UnsafeStorageFromTH off Type

gh-metadata: pytorch pytorch 21923 gh/li-roy/28/head

Commit:
Update on "Move UnsafeTensorFromTH and UnsafeStorageFromTH off Type"

Move UnsafeTensorFromTH and UnsafeStorageFromTH off Type

gh-metadata: pytorch pytorch 21923 gh/li-roy/28/head

Commit:
Update on "Move UnsafeTensorFromTH and UnsafeStorageFromTH off Type"

Move UnsafeTensorFromTH and UnsafeStorageFromTH off Type

gh-metadata: pytorch pytorch 21923 gh/li-roy/28/head

Commit:
Update on "Move UnsafeTensorFromTH and UnsafeStorageFromTH off Type"

Move UnsafeTensorFromTH and UnsafeStorageFromTH off Type

gh-metadata: pytorch pytorch 21923 gh/li-roy/28/head
